google/cloud/internal/rest_carrier.cc

```diff
@@ -0,0 +1,46 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifdef GOOGLE_CLOUD_CPP_HAVE_OPENTELEMETRY
+
+#include "google/cloud/internal/rest_carrier.h"
+#include "google/cloud/internal/noexcept_action.h"
+#include "google/cloud/internal/rest_context.h"
+
+namespace google {
+namespace cloud {
+namespace rest_internal {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+
+opentelemetry::nostd::string_view RestCarrier::Get(
+    opentelemetry::nostd::string_view) const noexcept {
+  // Since the client is never extracting data from the REST headers, we are not
+  // implementing Get and returning an empty string.
```

Comment:
>> perhaps we should assert that it is never called.
> 
> What do you mean by this?

```
  GCP_LOG(FATAL) << __func__ << " should never be called";
```

Reply:
You might additionally consider returning a default constructed (null) `string_view` instead of an empty one.

Reply:
Done and done

Reply:
"returning an empty string" in the comment was never truly accurate (it is a string view), but now that it is a null string view instead of an empty one, it probably deserves a little re-wording.